### ISSUE TITLE
Request: default axes view on bottom left of viewer canvas

### ISSUE THREAD

**Issue by jni** [napari/napari#1887]
## 🚀 Feature

In #1719, there was a lot of discussion that we should display axes as a little 3D compass on the bottom left of the image. That was left for a future PR but I don't think there's an issue about it. (?) This is that issue.

During a user interview, @DragaDoncila @AbigailMcGovern and I encountered a hilarious problem with having the axes at the origin, because their dataset in real coordinates was at translation offset (3000, -2000, -2000). One had to basically go to the moon, in relative terms, to see the axes visual. :joy:

Code to reproduce:

```python
import numpy as np
import napari
from skimage import data

blobs = data.binary_blobs(
        length=128, n_dim=3, volume_fraction=0.3
        ).astype(np.uint8) * 255

with napari.gui_qt():
    viewer = napari.view_image(
            blobs, ndisplay=3, translate=(3000, -2000, -2000)
            )
    viewer.camera.zoom = 0.1
    viewer.axes.visible = True
```

<img width="1200" alt="Screen Shot 2020-11-17 at 6 15 23 pm" src="https://user-images.githubusercontent.com/492549/99357916-df6af680-2900-11eb-8604-03ef8adeda37.png">

(I do have to note that the camera and axes API is super nice now!)

**Comment by sofroniewn** [comment]
Agreed. This was one of the enhancements requested in #1726, fine to make its own issue for it and bump it up the priority list!

**Comment by psobolewskiPhD** [comment]
So currently Axes are a SceneOverlay, which puts them in the data space, like the bounding box overlay.
One could switch them to a CanvasOverlay, then they'd be like the scale bar overlay, but then I'm not sure how that would work with 3D.

**Comment by Modjular** [comment]
Is`napari/_vispy/overlays/axes.py` a good place to start?

**Comment by TimMonko** [comment]
@Modjular that's definitely a good starting place! Peter's comment provides some good insight. Thinking about implementation more, I would move it to a CanvasOverlay, which will also tile nicely once we have new updates. I could see some issues continuing to use it as a SceneOverlay and using the recommended corner of the bounding box. 

I think my initial attempt would be to attach the camera to a CanvasOverlay and use the ScaleBar overlay as an example for this, where we connect to camera events (in this case the zoom, to adjust the scale). See: https://github.com/napari/napari/blob/4075179c23e76a89a54d956dbc5dd91131cd14b0/src/napari/_vispy/overlays/scale_bar.py#L41-L42

So, I would try connecting to `self.viewer.camera.events.angles` for rotating the axes.
Perhaps one tricky part will be determing the position of the axes because rotation will cause them to change the visual extent.

**Comment by brisvag** [comment]
Yep, definitely how I would do it too! I tried in the past and then got stuck somewhere in the weeds, but feel free to reach out or tag me @Modjular, i can help :)

**Comment by Modjular** [comment]
@brisvag Thanks for the offer! I have a couple questions:
 - As a first down, I'm trying to change the canvas overlay from 2D to 3D. Where is that set?
 - Is there a napari UML diagram anywhere I could reference? I'm having a lot of trouble figuring out if I'm looking at model or viewer or controller code.

**Comment by TimMonko** [comment]
> * Is there a napari UML diagram anywhere I could reference? I'm having a lot of trouble figuring out if I'm looking at model or viewer or controller code.

Just a quick driveby to try to be helpful. We have the architecture guide which describes how our MVC _aspires_ to be. https://napari.org/dev/developers/architecture/index.html (you may find the "UI Sections" part helpful for a mermaid representation of some things)
However, sometimes things are a bit mixed up. It's at least a start 🤷

**Comment by brisvag** [comment]
Yeah no proper diagram for now, sorry 😅 

Nothing prevents you from using a 3D visual as a canvas overlay (meaning, you don't need to specifically "change it" to 3D). You could start by trying to visualize a cube or something like that, and it should work out of the box.

**Comment by Modjular** [comment]
@brisvag  Is it wrong to say `VispyCanvasOverlay` acts as a 'billboard' that always remains pinned to its `CanvasPosition`? And that I shouldn't touch it's `self.node.transform`?

Also I'm confused by this line at the end of `VispyCanvasOverlay._on_position_change`

https://github.com/napari/napari/blob/4075179c23e76a89a54d956dbc5dd91131cd14b0/src/napari/_vispy/overlays/base.py#L125

Is this for `scale_bar`'s sake?

**Comment by brisvag** [comment]
I was also recently confused about that line (anmd yes, probably it's because of scale bar), but that's being removed in #7836, which is getting close to ready. You might actually be better of working from that branch, it should make things easier for you once that gets merged.

It's not quite a billboard because it's not necessarily 2D and facing the camera, but it does remain pinned to the canvas position. Basically, do not touch the *translate* part of the transform, which is being ussed to position the overlay in the appropriate corner. However, you can use scale and rotate just fine, and I *think* it should work as you expect :P

**Comment by Modjular** [comment]
@brisvag Thanks for the headsup!
I've got the rotations going but I'm encountering a rendering bug.

https://github.com/user-attachments/assets/b9b45dde-c0d1-4407-b75e-76cb86e78ece

I'm pretty `vispy` illiterate still, but I _think_ i'm encountering z-fighting since the overlays are set to `z = 0`. Any recommendations? Tweaking `z` to `-1` didn't help. I also suspect it might be the clipping plane.

**Comment by TimMonko** [comment]
@Modjular this is so awesome to see working! 
Would be amazing if you made a PR to our repo, seeing the code (especially in this quite working state!) will help us to troubleshoot this better. Super excited 😁

**Comment by Modjular** [comment]
@TimMonko Great idea. The headaches lies in `src/napari/_vispy/overlays/zyx_axes.py`.
Should discussion stay here or move to the PR?

**Comment by jni** [comment]
Yeah please open a draft PR @Modjular and we can work on the implementation together! Beyond excited to see this taking shape after all these years! 😍

**Comment by jni** [comment]
oh wait I see the PR's open, #8262 😅

**Comment by psobolewskiPhD** [comment]
@jni Can we consider #8262 to close this?

Edit: oops misclicked!
